main_test.go

```diff
@@ -0,0 +1,83 @@
+package main
+
+import (
+	"github.com/oyugirachel/deck"
+	"testing"
+	
+)
+
+var (
+	ace   = deck.Card{Suit: deck.Spade, Rank: deck.Ace}
+	two   = deck.Card{Suit: deck.Spade, Rank: deck.Two}
+	three = deck.Card{Suit: deck.Spade, Rank: deck.Three}
+	four  = deck.Card{Suit: deck.Spade, Rank: deck.Four}
+	five  = deck.Card{Suit: deck.Spade, Rank: deck.Five}
+	six   = deck.Card{Suit: deck.Spade, Rank: deck.Six}
+	seven = deck.Card{Suit: deck.Spade, Rank: deck.Seven}
+	eight = deck.Card{Suit: deck.Spade, Rank: deck.Eight}
+	nine  = deck.Card{Suit: deck.Spade, Rank: deck.Nine}
+	ten   = deck.Card{Suit: deck.Spade, Rank: deck.Ten}
+	jack  = deck.Card{Suit: deck.Spade, Rank: deck.Jack}
+	queen = deck.Card{Suit: deck.Spade, Rank: deck.Queen}
+	king  = deck.Card{Suit: deck.Spade, Rank: deck.King}
+)
+
+func Test_scoring(t *testing.T) {
+
+	type args struct {
+		snap bool
+	}
+	tests := []struct {
+		name  string
+		cards [2]deck.Card
+		args  args
+		score int
+	}{
+		{
+			name: "differentCards-snap",
+
+			cards: [2]deck.Card{six, ten},
+			args:  args{snap: true},
+			score : -1,
+		},
+		{
+			name:  "differentCards",
+			cards: [2]deck.Card{six, ten},
+			args:  args{snap: false},
+			score: 0,
+			
+		},
+
+		{
+			name:  "sameCards-snap",
+			cards: [2]deck.Card{ace, ace},
+			args:  args{snap: true},
+			score: 1,
+		},
+		{
+			name:  "sameCards",
+			cards: [2]deck.Card{ten, ten},
+			args:  args{snap: false},
+			score: -1,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			cards := deck.New(deck.Deck(1), deck.Shuffle)
+			lastCard = 1
+			presentCards[0] = presentCards[1]
+			presentCards[1] = cards[lastCard]
+			presentCards = tt.cards
```

Comment:
This line sets the value of presentCards, so you shouldn't need to use the lines above.

Check carefully what the scoring function uses. Does it depend on `lastCard` or `cards` at all? Do you need to set those up in this test?